Description: Upstream changes introduced in version 7.1.1-1
 This patch has been created by dpkg-source during the package build.
 Here's the last changelog entry, hopefully it gives details on why
 those changes were made:
 .
 pytango (7.1.1-1) unstable; urgency=low
 .
   * Initial release (Closes: #nnnn)  <nnnn is the bug number of your ITP>
 .
 The person named in the Author field signed this changelog entry.
Author: Picca Frédéric-Emmanuel <picca@synchrotron-soleil.fr>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- pytango-7.1.1.orig/setup.py
+++ pytango-7.1.1/setup.py
@@ -1,4 +1,4 @@
-import os, sys
+import os, sys, platform
 
 from distutils.core import setup, Extension
 from distutils.dist import Distribution
@@ -159,7 +159,6 @@ else:
     
     include_dirs += [ os.path.join(BOOST_ROOT, 'include') ]
     libraries += [
-        'boost_python',
         'pthread',
         'rt',
         'dl',
@@ -168,6 +167,15 @@ else:
         'omnithread',
         'COS4',
     ]
+    # when building with multiple version of python on debian we need
+    # to link against boost_python-py25/-py26 etc...
+    if platform.dist()[0] in ['debian']:
+        if distutils.sysconfig.get_python_version() == '2.5':
+            libraries += ['boost_python-py25']
+        elif distutils.sysconfig.get_python_version() == '2.6':
+            libraries += ['boost_python-py26']
+    else:
+        libraries += ['boost_python']
 
     # Note for PyTango developers:
     # Compilation time can be greatly reduced by compiling the file
